.github/workflows/build-iso.yml

```diff
@@ -0,0 +1,122 @@
+          echo "${ENDLESSCLIENT_CA_CHAIN}" >> client.crt
+          echo "${ENDLESSCLIENT_KEY}" > client.key
+          chmod 600 client.key
+
+      - name: Install BuildStream and dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            python3-pip python3-venv bubblewrap \
+            lzip xz-utils bzip2 gzip \
+            git wget curl \
+            build-essential pkg-config \
+            autotools-dev automake autoconf libtool \
+            cmake ninja-build meson \
+            flex bison gettext \
+            libssl-dev libffi-dev \
+            ostree flatpak-builder
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install BuildStream buildstream-plugins dulwich requests tomlkit
+
+      - name: Configure BuildStream
+        run: |
+          source venv/bin/activate
+          mkdir -p ~/.config/buildstream
+          cat > ~/.config/buildstream/user.conf << EOF
+          # BuildStream user configuration
+          scheduler:
+            fetchers: 4
+            builders: 2
+            pushers: 2
+            network-retries: 2
+          build:
+            # Allow up to 4 parallel processes to execute within the scope of one build
+            max-jobs: 4
+          logging:
+            key-length: 0
+            verbose: false
+            error-lines: 20
+            message-lines: 20
+            debug: false
+            element-format: |
+              %{state: >12} %{full-key} %{name} %{workspace-dirs}
+            message-format: |
+              [%{elapsed}][%{key}][%{element}] %{action} %{message}
+          EOF
+
+      - name: Generate secure boot keys
+        run: |
+          make -C files/boot-keys clean
+          make -C files/boot-keys IMPORT_MODE=snakeoil
+
+      - name: Configure BuildStream artifact cache
+        run: |
+          source venv/bin/activate
+          # Configure BuildStream to use the artifact cache with client certificates
+          cat >> ~/.config/buildstream/user.conf << EOF
+
+          # Artifact cache configuration
+          artifacts:
+            - url: https://bstcache.endlessos.org/
+              push: true
+              client-cert: $(pwd)/client.crt
+              client-key: $(pwd)/client.key
+          EOF
+
+      - name: Enable unprivileged userns
+        run: |
+          source venv/bin/activate
+          sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
+
+      - uses: runs-on/action@v2
+        with:
+          sccache: s3
+      - uses: mozilla-actions/sccache-action@v0.0.9
+      - name: Build ISO image
+        run: |
+          source venv/bin/activate
+          bst build gnome-build-meta.bst:iso/image.bst
+
+      - name: Push artifacts to cache
+        run: |
+          source venv/bin/activate
+          # Push all built artifacts to the cache server
+          bst artifact push --deps all gnome-build-meta.bst:iso/image.bst
```

Comment:
I think it would be good to run this step all the time so that we can get stuff cached even if the build fails at some point. See the [always](https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#always) expression, although it may be better to take their suggestion and use `if: ${{ !cancelled() }}`.